tests/test_modeling_electra.py

```diff
@@ -0,0 +1,287 @@
+                input_ids, attention_mask=input_mask, token_type_ids=token_type_ids, labels=token_labels
+            )
+            result = {
+                "loss": loss,
+                "logits": logits,
+            }
+            self.parent.assertListEqual(
+                list(result["logits"].size()), [self.batch_size, self.seq_length, self.num_labels]
+            )
+            self.check_loss_output(result)
+
+        def create_and_check_electra_for_pretraining(
+            self,
+            config,
+            input_ids,
+            token_type_ids,
+            input_mask,
+            sequence_labels,
+            token_labels,
+            choice_labels,
+            fake_token_labels,
+        ):
+            config.num_labels = self.num_labels
+            model = ElectraForPreTraining(config=config)
+            model.to(torch_device)
+            model.eval()
+            loss, logits = model(
+                input_ids, attention_mask=input_mask, token_type_ids=token_type_ids, labels=fake_token_labels
+            )
+            result = {
+                "loss": loss,
+                "logits": logits,
+            }
+            self.parent.assertListEqual(list(result["logits"].size()), [self.batch_size, self.seq_length])
+            self.check_loss_output(result)
+
+        def prepare_config_and_inputs_for_common(self):
+            config_and_inputs = self.prepare_config_and_inputs()
+            (
+                config,
+                input_ids,
+                token_type_ids,
+                input_mask,
+                sequence_labels,
+                token_labels,
+                choice_labels,
+                fake_token_labels,
+            ) = config_and_inputs
+            inputs_dict = {"input_ids": input_ids, "token_type_ids": token_type_ids, "attention_mask": input_mask}
+            return config, inputs_dict
+
+    def setUp(self):
+        self.model_tester = ElectraModelTest.ElectraModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=ElectraConfig, hidden_size=37)
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_electra_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_electra_model(*config_and_inputs)
+
+    def test_for_masked_lm(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_electra_for_masked_lm(*config_and_inputs)
+
+    def test_for_token_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_electra_for_token_classification(*config_and_inputs)
+
+    def test_for_pre_training(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_electra_for_pretraining(*config_and_inputs)
+
+    @slow
```

Comment:
(All IMO and inconsistent with ways other models are tested so feel free to disregard)

1) To me if you are going to pay the cost of downloading from S3+instantiating, you might want to check that the models fill in masks or do some other task in a reasonable way. In BART land, this has made it a lot easier to review other peoples changes -- since I know they don't break summarization/classification if my `@slow` tests pass.  I also found bugs a number of bugs that way.

2) How long does this take? do I really need to download 7 models to make sure that something I changed didn't break your code? Recall that we have `check_repository_consistency` sniffing out bad links.  

To me we'd get more from one slow test that checks whether `ElectraForMaskedLM` fills in a mask correctly than this slower test that basically only covers `__init__` (not even loading_info).

I guess if we were running doctests with more regularity that would be another way to get that kind of coverage.

Feel free to disregard!

Reply:
I agree with that; integration testing would be better. I'm updating the code.

Reply:
Will work on that in the following ELECTRA PR.